### PR TITLE
Fix mechanism to pass flag for leak tracking.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/flutter/test/flutter_test_config.dart [hunk 1]
```diff
@@ -46,6 +50,13 @@ Future<void> testExecutable(FutureOr<void> Function() testMain) {
       .withIgnored(
         allNotGCed: true,
       );
+
+    // Print is here in spite of
+    // https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#only-log-actionable-messages-to-the-console
+    // to provide a way to detect if the env variable is passed as expected on bots, that is not obvious.
+    // Sometimes the bot's environment contains `LEAK_TRACKING=true` but the value is not passed to the test.
+    // https://github.com/dart-lang/sdk/issues/54568
+    debugPrint('Leak tracking is enabled.');
```
Comment (keyonghan) [inline packages/flutter/test/flutter_test_config.dart:59]: Could you validate this by `export LEAK_TRACKING=true`, and run the test locally? If validated, then no need to add this print.

Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart:59]: Nope, it does not work locally: 

```
polinach-macbookpro3:flutter polinach$ echo $LEAK_TRACKING
true
polinach-macbookpro3:flutter polinach$  flutter test /Users/polinach/_/flutter_dev/packages/flutter/test/widgets/page_view_test.dart
00:03 +33: All tests passed!                                                                               
polinach-macbookpro3:flutter polinach$  flutter test --dart-define LEAK_TRACKING=true /Users/polinach/_/flutter_dev/packages/flutter/test/widgets/page_view_test.dart
00:01 +0: loading /Users/polinach/_/flutter_dev/packages/flutter/test/widgets/page_view_test.dart                                                                              
Leak tracking is enabled.
```

Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart:59]: But it works on bots

Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart:59]: That's why I filed a bug: https://github.com/dart-lang/sdk/issues/54568

Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart:59]: Refiled to flutter: https://github.com/flutter/flutter/issues/141299

Comment (keyonghan) [reply packages/flutter/test/flutter_test_config.dart:59]: Hum, weird, it works for me locally with your patch `const bool.fromEnvironment('LEAK_TRACKING') || (bool.tryParse(Platform.environment['LEAK_TRACKING'] ?? '')  ?? false);`:

```
keyonghan-macbookpro2:test keyonghan$ echo $LEAK_TRACKING
true
keyonghan-macbookpro2:test keyonghan$ flutter test /Users/keyonghan/Documents/work/flutter/packages/flutter/test/widgets/page_view_test.dart
Changing current working directory to: /Users/keyonghan/Documents/work/flutter/packages/flutter
00:01 +0: loading /Users/keyonghan/Documents/work/flutter/packages/flutter/test/widgets/page_view_test.dart                                                                                                 
Leak tracking is enabled.
```

Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart:59]: Weird

Added version information to the bug.

Comment (keyonghan) [reply packages/flutter/test/flutter_test_config.dart:59]: Nit: add a TODO to remove after the reference issue is fixed.

**Review by keyonghan — APPROVED** [review]
LGTM with nit.

File: packages/flutter/test/flutter_test_config.dart (hunk 1 above)
Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart]: @stuartmorgan found what is wrong: https://github.com/flutter/flutter/issues/141299

Removed print.

**Pull request merged**